spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/BaseTempFileS3OutputStream.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.s3;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+/**
+ * {@link BaseTempFileS3OutputStream} abstract class defining the common behaviour for implementations that use a temp
+ * file for the {@link S3OutputStream}. Extracted from the original implementation of
+ * {@link DiskBufferingS3OutputStream} by Maciej Walkowiak
```

Comment:
```suggestion
 * {@link BaseTempFileS3OutputStream} defines the common behaviour for implementations that use a temporary
 * file for the {@link S3OutputStream}.
```

Just add me below as a second author